packages/netsuite-adapter/src/change_validators/remove_list_item_without_scriptid.ts

```diff
@@ -0,0 +1,187 @@
+import { PERMISSIONS, ROLE } from '../constants'
+
+const log = logger(module)
+
+const PERMISSION = 'permission'
+
+type RolePermissionObject = {
+  permkey: string | ReferenceExpression
+  permlevel: string
+  restriction?: string
+}
+
+export type ItemInList = RolePermissionObject
+
+type GetItemList = (instance: InstanceElement) => ItemInList[]
+type GetItemString = (item: ItemInList) => string
+type GetItemByID = (instance: InstanceElement, id: string) => ItemInList | undefined
+type GetListPath = () => string[]
+type GetMessage = (removedListItems: string[]) => string
+
+const getMessageByElementNameAndListItems = (elemName: string, removedListItems: string[]): string =>
+  `Can't remove the inner ${elemName}${removedListItems.length > 1 ? 's' : ''} ${removedListItems.join(', ')}. NetSuite supports the removal of inner elements only from its UI.`
+
+export type ItemListGetters = {
+  getItemList: GetItemList
+  getItemString: GetItemString
+  getItemByID: GetItemByID
+  getListPath: GetListPath
+  getDetailedMessage: GetMessage
+}
+
+const isRolePermissionObject = (obj: Value): obj is RolePermissionObject => {
+  const returnVal = (
+    _.isPlainObject(obj)
+    && (typeof obj.permkey === 'string' || isReferenceExpression(obj.permkey))
+    && typeof obj.permlevel === 'string'
+    && (typeof obj.restriction === 'string' || obj.restriction === undefined)
+  )
+  if (!returnVal) {
+    log.debug('There is a role permission with a different shape', obj)
+  }
+  return returnVal
+}
+
+const getRoleListPath: GetListPath = () => [
+  PERMISSIONS,
+  PERMISSION,
+]
+
+const getRolePermissionList:GetItemList = instance => {
+  const listPathValue = _.get(instance.value, getRoleListPath())
+  if (_.isPlainObject(listPathValue)) {
+    return Object.values(listPathValue)
+      .filter(isRolePermissionObject)
+  }
+  if (listPathValue !== undefined) {
+    log.debug("Role permissions wasn't a plain object under permissions.permission", instance)
+  }
+  return []
+}
+
+const getRolePermkey: GetItemString = (
+  permission: RolePermissionObject
+): string => {
+  const { permkey } = permission
+  if (_.isString(permkey)) {
+    return permkey
+  }
+  return permkey.value
+}
+
+const getRolePermissionByName = (
+  role: InstanceElement,
+  id: string,
+): RolePermissionObject | undefined => (
+  getRolePermissionList(role)
+    .find(permObj => getRolePermkey(permObj) === id)
+)
+
+const getRoleMessage = (removedListItems: string[]): string =>
+  getMessageByElementNameAndListItems(PERMISSION, removedListItems)
+
+const roleGetters: ItemListGetters = {
+  getItemList: getRolePermissionList,
+  getItemString: getRolePermkey,
+  getItemByID: getRolePermissionByName,
+  getListPath: getRoleListPath,
+  getDetailedMessage: getRoleMessage,
+}
+
+export const getGettersByType = (
+  typeName: string,
+): ItemListGetters | undefined => {
+  if (typeName === ROLE) {
+    return roleGetters
+  }
+  return undefined
+}
+const getIdentifierList = (
```

Comment:
styling nit - empty line between functions
```suggestion
}

const getIdentifierList = (
```